pkgs/development/python-modules/playwright/default.nix

```diff
@@ -0,0 +1,84 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, greenlet
+, websockets
+, pyee
+, pillow
+, pixelmatch
+, autobahn
+, pyopenssl
+, service-identity
+, requests
+, objgraph
+, callPackage
+}:
+
+buildPythonPackage rec {
+  pname = "playwright";
+  version = "1.22.0";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "playwright-python";
+    rev = "v${version}";
+    sha256 = "sha256-nodm1Wts61vRCqjqNX54tD0cQB7uoxx6Eypz7nBR54E=";
+  };
+
+  format = "pyproject";
+
+  playwright-driver = callPackage ./driver.nix { playwrightVersion = version; };
+
+  postPatch = ''
+    # works with pyee 9 as well, which is what nixos has packaged
+    # disables setuptools-scm, as it fails
+    substituteInPlace meta.yaml --replace "- pyee ==8.1.0" ""
+    substituteInPlace setup.py \
+        --replace '"pyee==8.1.0",' "" \
+        --replace '"setuptools-scm==6.3.2",' ""
+    # replace functionality that setuptools-scm provides by adding the version to a file
+    echo 'version = "${version}"' > playwright/_repo_version.py
+  '';
+
+  preBuild = ''
+    mkdir -p driver
+    cp -r ${playwright-driver}/driver ./
+  '';
+
+  preInstall = ''
+    # As pip names the wheel "dist", but when installing it will expect a properly formatted filename in dist
+    mv dist dist.orig
+    mkdir -p dist
+    mv dist.orig dist/playwright-${version}-py3-none-any.whl
+  '';
+
+  propagatedBuildInputs = [
+    playwright-driver
+    greenlet
+    websockets
+    pyee
+    pillow
+    pixelmatch
+    autobahn
+    pyopenssl
+    service-identity
+    requests
+    objgraph
+  ];
+
+  # skip tests because they require network access
+  doCheck = false;
+
+  pythonImportsCheck = [
+    "playwright"
+  ];
+
+  meta = with lib; {
+    description = "Python version of the Playwright testing and automation library";
+    homepage = "https://github.com/microsoft/playwright-python/";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ techknowlogick ];
```

Comment:
```suggestion
    maintainers = with maintainers; [ techknowlogick ];
    platforms = platforms.linux ++ platforms.darwin;
```

Reply:
Are you [sure](https://github.com/microsoft/playwright-python/blob/main/setup.py#L79)?

Reply:
Good point. But seeing `drivers.nix` only supports Linux and Mac its probably best to keep the platforms setting like this, right? Not sure how practical it is to add the Windows binaries…

Reply:
I meant this covers some platforms that aren't in that upstream file:
```nix
nix-repl> playwright.meta.platforms
[ "aarch64-linux" "armv5tel-linux" "armv6l-linux" "armv7a-linux" "armv7l-linux" "i686-linux" "m68k-linux" "mipsel-linux" "mips64el-linux" "powerpc64-linux" "powerpc64le-linux" "riscv32-linux" "riscv64-linux" "s390-linux" "s390x-linux" "x86_64-linux" "x86_64-darwin" "i686-darwin" "aarch64-darwin" "armv7a-darwin" ]
```

Reply:
Oh yeah, then we probably want `[ "aarch64-linux" "x86_64-linux" "aarch64-darwin" "x86_64-darwin" ]`.

Reply:
updated, thanks :)